docs/src/main/paradox/couchbase.md

```diff
@@ -20,164 +30,117 @@ The table below shows direct dependencies of this module and the second tab show
 
-Scala
-: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #init-BulkSink }
 
-Java
-: @@snip (/couchbase/src/test/java/docs/javadsl/Examples.java) { #upsertBulk }
+### Get by ID
 
-We can create `CouchbaseSink` for single delete operation
+`CouchbaseFlow.fromId` methods allow to read documents specified by the document ID in the Akka Stream.
 
 Scala
-: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #delete-SingleSink }
+: @@snip [snip](/couchbase/src/test/scala/docs/scaladsl/CouchbaseFlowSpec.scala) { #fromId }
 
 Java
-: @@snip (/couchbase/src/test/java/docs/javadsl/Examples.java) { #delete-single-sink }
-
-There is also an option to create Sink that deletes bulk of documents by their Ids
-
-Scala
-: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #delete-BulkSink }
-
-Java
-: @@snip (/couchbase/src/test/java/docs/javadsl/Examples.java) { #delete-bulk-sink }
-
-#Flow Usage
+: @@snip [snip](/couchbase/src/test/java/docs/javadsl/CouchbaseExamplesTest.java) { #fromId }
 
-Couchbase Flow implements exactly the same method for upsert and delete as `CouchbaseSink`, but unlike Sink, an output of the Flow Stage is the following:
 
-- For an operation on single object output will be instance of `SingleOperationResult` class which consists of input `T` and optional exception of failed operation 
+## Writing to Couchbase in Akka Streams
 
-- For an operation on bulk the output will be instance of `BulkOperationResult` class which consists of Collection of input elements and Collection failed documents Id's and its relevant Exception. 
+For each mutation operation we need to create `CouchbaseWriteSettings` instance which consists of the following parameters
 
-This is vivid if you want to handle partial or single failures and do not want to affect stream materialization.  
+- Parallelism in access to Couchbase (default 1)
+- Couchbase Replication Factor (default `ReplicateTo.NONE`) 
+- Couchbase Persistence Level for Write Operation (default `PersistTo.NONE`)
+- 2 seconds operation timeout 
 
-Creating flow with single upsert operation
+These default values are not recommended for real use, as they do not observe persistence on disk for any node. 
 
 Scala
-: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #init-SingleFlow }
+: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseFlowSpec.scala) { #write-settings }
 
 Java
-: @@snip (/couchbase/src/test/java/docs/javadsl/Examples.java) { #upsertFlowSingle }
+: @@snip [snip](/couchbase/src/test/java/docs/javadsl/CouchbaseExamplesTest.java) { #write-settings }
 
-Creating flow for delete Bulk operation
 
-Scala
-: @@snip (/couchbase/src/test/scala/docs/scaladsl/CouchbaseSupport.scala) { #delete-BulkFlow }
+Read more about durability settings in the [Couchbase documentation](https://docs.couchbase.com/java-sdk/2.6/durability.html#configuring-durability). 
```

Comment:
We depend on couchbase `2.7.2` but here link to the `2.6` docs.

Reply:
I changed all the links to use `extref` now.